psutil/tests/__init__.py

```diff
@@ -468,6 +448,10 @@ def pyrun(src, **kwds):
     kwds.setdefault("stdout", None)
     kwds.setdefault("stderr", None)
     srcfile = get_testfn()
+    if sys.executable != PYTHON_EXE:
+        env = kwds.get("env", os.environ).copy()
+        env["__PYVENV_LAUNCHER__"] = sys.executable
+        kwds["env"] = env
```

Comment:
Please add a comment explaining why this is necessary. I am sure that I will look a this in X months and will be clueless. :)